components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java

```diff
@@ -113,15 +123,43 @@ public synchronized static APIGatewayManager getInstance() {
                 continue;
             }
             APIGatewayAdminClient client;
+            LocalEntryAdminClient localEntryAdminClient;
+
             try {
-                client = new APIGatewayAdminClient(environment);
+                String definition;
                 String operation;
+                client = new APIGatewayAdminClient(environment);
                 long apiGetStartTime = System.currentTimeMillis();
                 APIData apiData = client.getApi(tenantDomain, api.getId());
                 endTime = System.currentTimeMillis();
                 if (debugEnabled) {
                     log.debug("Time taken to fetch API Data: " + (endTime - apiGetStartTime) / 1000 + "  seconds");
                 }
+                localEntryAdminClient = new LocalEntryAdminClient(environment, tenantDomain);
+                if (api.getType() != null && api.getType().equals(APIConstants.APITransportType.GRAPHQL.toString())) {
+                    //Build schema with scopes and roles
+                    definition = buildSchemaWithScopesAndRoles(api);
+                    localEntryAdminClient.deleteEntry(api.getUUID() + "_graphQL");
+                    localEntryAdminClient.addLocalEntry("<localEntry key=\"" + api.getUUID() + "_graphQL" + "\">" +
+                         definition + "</localEntry>");
+
+                    Set<URITemplate> uriTemplates = new HashSet<>();
+                    URITemplate template = new URITemplate();
+                    template.setAuthType("Any");
+                    template.setHTTPVerb("POST");
+                    template.setHttpVerbs("POST");
+                    template.setHttpVerbs("GET");
+                    template.setUriTemplate("/*");
+                    uriTemplates.add(template);
+                    api.setUriTemplates(uriTemplates);
+                } else {
+                    definition = api.getSwaggerDefinition();
+                    localEntryAdminClient.deleteEntry(api.getUUID());
+                    localEntryAdminClient.addLocalEntry("<localEntry key=\"" + api.getUUID() + "\">" +
+                                definition.replaceAll("&(?!amp;)", "&amp;").
```

Comment:
can't we combine this line and the 2 lines into 2 lines instead of 3?

Reply:
can't combine this expression to two lines